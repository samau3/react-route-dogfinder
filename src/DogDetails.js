import React from "react";
import { useParams } from "react-router-dom";

function DogDetails({ dogs }) {

    const { name } = useParams();

    console.log("DogDetails - useParams()", useParams());
    console.log("DogDetails - name:", { name });
    console.log("DogDetails - dogs:", { dogs });

    const requestedDog = dogs.find(dog =>
        dog.name.toLowerCase() === name)

    console.log("DogDetails - requestedDog:", { requestedDog });

    return (
        <div>
            <p>Name: {requestedDog.name}</p>
            <img src={`../${requestedDog.src}.jpg`} alt={`${requestedDog.src}`} />
            <p>Age: {requestedDog.age} </p>
            <p>Facts:
                <ul>
                    {requestedDog.facts.map(fact => (
                        <li key={fact}>
                            Fact: {fact}</li>
                    ))}
                </ul>
            </p>
        </div>
    )
}

export default DogDetails;