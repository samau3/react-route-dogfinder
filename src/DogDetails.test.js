import React from 'react';
import DogDetails from './DogDetails';

import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { TEST_DOGS } from './_testCommon';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // don't want to mock all library
    useParams: () => ({ name: "whiskey" }),
}))

// basic tests
it('renders without crashing', function () {

    render(
        <MemoryRouter>
            <DogDetails dogs={TEST_DOGS} />
        </MemoryRouter>
    );
});

it('matches snapshot', function () {
    const container = render(
        <MemoryRouter>
            <DogDetails dogs={TEST_DOGS} />
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});
// end basic tests

it('check if dog detail loaded properly', function () {
    const { container } = render(
        <MemoryRouter>
            <DogDetails dogs={TEST_DOGS} />
        </MemoryRouter>
    );
    const facts = container.querySelectorAll("ul li");
    expect(facts).toHaveLength(3);
});