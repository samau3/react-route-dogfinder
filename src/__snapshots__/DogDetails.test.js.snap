// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      Name: Whiskey
    </p>
    <img
      alt="whiskey"
      src="../whiskey.jpg"
    />
    <p>
      Age: 5 
    </p>
    Facts:
    <ul>
      <li>
        Fact: Whiskey loves eating popcorn.
      </li>
      <li>
        Fact: Whiskey is a terrible guard dog.
      </li>
      <li>
        Fact: Whiskey wants to cuddle with you!
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
